Exam-Service/src/main/java/tw/waterball/judgegirl/examservice/usecases/GetExamUseCase.java

```diff
@@ -4,22 +4,32 @@
 import lombok.Data;
 import tw.waterball.judgegirl.commons.exceptions.NotFoundException;
 import tw.waterball.judgegirl.entities.Exam;
+import tw.waterball.judgegirl.entities.Question;
 import tw.waterball.judgegirl.examservice.repositories.ExamRepository;
+import tw.waterball.judgegirl.problemapi.clients.ProblemServiceDriver;
+import tw.waterball.judgegirl.problemapi.views.ProblemView;
 
 import javax.inject.Named;
 
 @Named
 @AllArgsConstructor
 public class GetExamUseCase {
     private final ExamRepository examRepository;
+    private final ProblemServiceDriver problemServiceDriver;
 
     public void execute(Request request, Presenter presenter) {
         Exam exam = examRepository.findById(request.examId).orElseThrow(NotFoundException::new);
         presenter.setExam(exam);
+        for (Question question : exam.getQuestions()) {
+            ProblemView problemView = problemServiceDriver.getProblem(question.getId().getProblemId());
+            presenter.addProblems(problemView);

```

Comment:
This is a very inefficient approach (fetch problem one by one)
but unfortunately, this requires our problem-service to support a new API to help fetch problems all once (given a list of problem's ids)  @nyngwang #151 issue created for this

mark a TODO ahead for the moment
`// TODO: should be improved to fetch all the problems in one query`

Reply:
add an issue for this